src/gd_avif.c

```diff
@@ -0,0 +1,667 @@
+/**
+ * File: AVIF IO
+ *
+ * Read and write AVIF images using libavif (https://github.com/AOMediaCodec/libavif) .
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <math.h>
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdhelpers.h"
+
+#ifdef HAVE_LIBAVIF
+#include "avif.h"
+#include "internal.h"
+// internal.h is only used for AVIF_MIN() and AVIF_CLAMP() macros
+
+/*
+  Define defaults for encoding images:
+    DEFAULT_CHROMA_SUBSAMPLING: 4:2:0 is commonly used for Chroma subsampling.
+    DEFAULT_MIN_QUANTIZER, DEFAULT_MAX_QUANTIZER: 
+      We need more testing to really know what quantizer settings are optimal,
+      but teams at Google have been using maximum=30 as a starting point.
+    DEFAULT_QUALITY: following gd conventions, -1 indicates the default.
+    DEFAULT_SPEED: AVIF_SPEED_DEFAULT is -1. This simply tells the AVIF encoder to use the default speed.
+*/
+
+#define DEFAULT_CHROMA_SUBSAMPLING AVIF_PIXEL_FORMAT_YUV420
+#define DEFAULT_QUANTIZER 30
+#define DEFAULT_QUALITY -1
+#define DEFAULT_SPEED AVIF_SPEED_DEFAULT
+
+// This initial size for the gdIOCtx is standard among GD image conversion functions
+#define NEW_DYNAMIC_CTX_SIZE 2048
+
+// Our quality param ranges from 0 to 100.
+// To calculate quality, we convert from AVIF's quantizer scale, which runs from 63 to 0.
+#define MAX_QUALITY 100
+
+// For computing the number of tiles and threads to use during encoding
+// Maximum threads are from libavif/contrib/gkd-pixbuf/loader.c
+#define MIN_TILE_AREA (512 * 512)
+#define MAX_TILES 6
+#define MAX_THREADS 64
+
+
+/*** Helper function signatures ***/
+
+static avifBool _gdImageAvifCtx (gdImagePtr im, gdIOCtx * outfile, int quality, int speed);
```

Comment:
omit spaces before `(` with funcs

and after `*` with pointers

Reply:
Many functions in the codebase have a space before the `(` and args... but I assume that you know that and want contributors to follow this convention going forward :)

spaces before left parens in function declarations removed.

Reply:
(and I think I got all the spaces after *'s for pointers yesterday; when I check this in they should be gone)

Reply:
yes, i noted somewhere else that we're inconsistent between the two forms.  so i'd like to move in this direction with new code.